packages/http-server/src/server.ts

```diff
@@ -10,63 +11,75 @@ import { IPrismDiagnostic } from '@stoplight/prism-core';
 import { pipe } from 'fp-ts/lib/pipeable';
 import * as TE from 'fp-ts/lib/TaskEither';
 import * as E from 'fp-ts/lib/Either';
+import { MicriHandler } from 'micri/dist';
+
+function searchParamsToNameValues(searchParams: URLSearchParams): IHttpNameValues {
+  const params = {};
+  for (const key of searchParams.keys()) {
+    const values = searchParams.getAll(key);
+    params[key] = values.length === 1 ? values[0] : values;
```

Comment:
Is this mandatory? Can't we use always an array?

Reply:
I believe it is. The array is returned only when there are multiple name-value pairs: `a=1&a=2&a=3`. Prism relies on this behavior inside at least the validator.

It is also part of a bigger problem. I think then the parsing should be done inside validators (then we call it a parser). Then we don't need to guess whether it should be an array of string or whatever. There would be also a single place which does the parsing. We learned already that parsing might be very complicated due to encodings and other stuff.